Source/Core/Core/Core.cpp

```diff
@@ -323,24 +326,24 @@ static void CpuThread(const std::optional<std::string>& savestate_path, bool del
   DeclareAsCPUThread();
 
   const SConfig& _CoreParameter = SConfig::GetInstance();
-
   if (_CoreParameter.bCPUThread)
-  {
     Common::SetCurrentThreadName("CPU thread");
-  }
   else
-  {
     Common::SetCurrentThreadName("CPU-GPU thread");
-    g_video_backend->Video_Prepare();
-    Host_Message(WM_USER_CREATE);
-  }
 
   // This needs to be delayed until after the video backend is ready.
   DolphinAnalytics::Instance()->ReportGameStart();
 
   if (_CoreParameter.bFastmem)
     EMM::InstallExceptionHandler();  // Let's run under memory watch
 
+  if (savestate_path)
+  {
+    ::State::LoadAs(*savestate_path);
```

Comment:
Is it really safe to have this before the call to `MemoryWatcher::Init`? That init function schedules a CoreTiming event.

Reply:
Good catch, thanks.